bundles/org.eclipse.jface/src/org/eclipse/jface/resource/ImageDescriptor.java

```diff
@@ -87,6 +91,21 @@ public static ImageDescriptor createFromFile(Class<?> location, String filename)
 		return new FileImageDescriptor(location, filename);
 	}
 
+	/**
+	 * Creates and returns a new image descriptor from a plug-in.
+	 *
+	 * @param location to determine the bundle which contains the file
+	 * @param filePath path to the file relative in the bundle
+	 * @return a new image descriptor
+	 * @since 3.32
+	 */
+	public static ImageDescriptor createFromBundle(Class<?> location, String filePath) {
```

Comment:
What happens if you want to load an image from a different bundle where you don't have a class?
In WindowBuilder, we have something like this, for example:

> ResourceManager.getPluginImage("org.eclipse.jdt.ui", "/icons/full/elcl16/ch_cancel.png");

Perhaps it makes to add a second (or even third) method:

````
ImageDescriptor createFromBundle(Class<?> location, String filePath) {
  return createFromBundle(FrameworkUtil.getBundle(location), String filePath);
}

ImageDescriptor createFromBundle(String symbolicName, String filePath) {
  return createFromBundle(Platform.getBundle(symbolicName), String filePath);
}

ImageDescriptor createFromBundle(Bundle bundle, String filePath) {
  ...
}
```